vault/resource_aws_secret_backend_role.go

```diff
@@ -70,6 +70,16 @@ func awsSecretBackendRoleResource() *schema.Resource {
 				Required:    true,
 				Description: "Role credential type.",
 			},
+			"role_arns": {
```

Comment:
Do `role_arns` conflict with other fields in practice? [The documentation](https://www.vaultproject.io/api/secret/aws/index.html#role_arns) doesn't say it does. Should we remove it from the conflicts with clauses?

Reply:
* `policy_document` apparently does not conflict with `role_arns` (see [test using both](https://github.com/joelthompson/vault/blob/f88c1b37b751a9add6ed3c0487fa9a0bf2bb0362/builtin/logical/aws/backend_test.go#L657-L658)), so I've removed the ConflictsWith clause for this one
* `policy_arns` and `policy_arn` are only valid when `credential_type` is `iam_user`, while `role_arns` is only valid when `credential_type` is `assumed_role` and prohibited otherwise (per [the doc](https://www.vaultproject.io/api/secret/aws/index.html#parameters-3)), so in practice, they are mutually exclusive
* for `policy`, the doc says "cannot be mixed with the parameters listed above", hence the `ConflictsWith` clause